src/vm/moar/QAST/QASTCompilerMAST.nqp

```diff
@@ -2509,30 +2509,37 @@ class MoarVM::BytecodeWriter {
         $!mbc.write_uint32($offset); # Offset of SC dependencies table
         $!mbc.write_uint32(my uint $num_sc_handles := nqp::elems(@sc_handles)); # Number of entries in SC dependencies table
         $offset := $offset + $sc_deps_size;
+        nqp::die("MAST::CompUnit file size limit reached!") if $offset < $sc_deps_size;
```

Comment:
How useful would it be to know which of these checks triggered? E.g., `
        nqp::die("MAST::CompUnit file size limit reached because of number of entries in SC dependencies table!") if $offset < $sc_deps_size;`

Reply:
Not sure it'll be terribly useful to an end user, especially since it may be a small thing that just barely pushes it over the limit.